### PR TITLE
Lindenbaum-Tarski algebra

### PR DESCRIPTION
**Pull request by croos90**
Lindenbaum-Tarski algebra of propostional logic.

- Definition of provability
- Properties of propositional logic
- Biprovability relation
- Definition of Lindenbaum-Tarski algebra
- Lindenbaum-Tarski algebra is complemented distributive lattice, hence Boolean

### CODE REVIEW

File: Cubical/Algebra/LindenbaumTarski.agda [hunk 1]
```diff
@@ -0,0 +1,391 @@
+{-# OPTIONS --safe #-}
```
Comment (mortberg) [inline Cubical/Algebra/LindenbaumTarski.agda:1]: Maybe add a short comment in the top of the file saying what it contains?

**Review by mortberg — APPROVED** [review]
Apart from a small comment I think this looks good. 

One could maybe split this into multiple files and move some stuff to other parts, like the definition of propositional formulas and the proof system... On the other hand, the logical system is somewhat tailored for the construction of the LT algebra.  What does @ecavallo  and @felixwellen think?

**Comment by felixwellen** [comment]
I think it's fine like it is now (agreeing that a comment explaining the file contents would be great). 

We can think about splitting off the logic-parts in our future refactoring that finally puts the logic related things in the library at a place where they are not completely unexpected ;-)

File: Cubical/Algebra/LindenbaumTarski.agda (hunk 1 above)
Comment (croos90) [reply Cubical/Algebra/LindenbaumTarski.agda:14]: Valid point!

File: Cubical/Algebra/LindenbaumTarski.agda [hunk 2]
```diff
@@ -0,0 +1,403 @@
+{-
+
+This file contains:
+
+  - Classical propositional logic and provability
+  - Properties of classical propositional logic
+  - Equivalence relation on formulas in terms of provability
+  - Definition of Lindenbaum-Tarski algebra
+  - Proof the Lindenbaum-Tarski algebra is a Boolean algebra
```
Comment (mortberg) [inline Cubical/Algebra/LindenbaumTarski.agda:9]:
```suggestion
  - Proof that the Lindenbaum-Tarski algebra is a Boolean algebra
```

File: Cubical/Algebra/LindenbaumTarski.agda (hunk 1 above)
Comment (mortberg) [reply Cubical/Algebra/LindenbaumTarski.agda:14]: Great!

**Review by mortberg — APPROVED** [review]
Merging as soon CI is finished checking it

**Pull request merged**